openedx/features/user_leads/helpers.py

```diff
@@ -0,0 +1,29 @@
+from django.db import IntegrityError, transaction
+
+from .constants import UTM_PARAM_NAMES
+from .models import UserLeads
+
+
+def get_utm_params(request):
+    request_get_params = request.GET
+    utm_params = {}
+
+    for key in UTM_PARAM_NAMES.values():
+        utm_param = request_get_params.get(key)
+        if utm_param:
+            utm_params.update({key: utm_param})
+
+    return utm_params
+
+
+def save_user_utm(request):
+    user = request.user
+    origin = request.resolver_match.url_name
+    utm_params = get_utm_params(request)
+
+    if not user.is_anonymous():
+        try:
+            with transaction.atomic():
+                UserLeads.objects.create(user=user, origin=origin, **utm_params)
```

Comment:
I'm just curious why we added this under the transaction as we are doing a single query. Isn't it?

Reply:
I was facing this issue when just simply passing on the exception: https://stackoverflow.com/questions/21458387/transactionmanagementerror-you-cant-execute-queries-until-the-end-of-the-atom